File: C++/728.cpp
__________________________________________________________________________________________________
sample 4 ms submission
class Solution {
public:
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int> ans;
        for(int i = left; i <= right; i++) {
            int rem = i;
            while (rem != 0) {
                int digit = rem % 10;
                if (digit == 0 || i % digit != 0) {
                    break;
                }
                else if (rem/10 == 0 && i % rem == 0) {
                    ans.push_back(i);
                }
                rem = rem/10;
            }
            
        }
        return ans;
    }
};
__________________________________________________________________________________________________
sample 8356 kb submission
class Solution {
public:
    bool isSelfDividing (int nr)
    {  int ctr = 1;
        do
        {
            int divisor = nr/ctr%10;
            if (divisor==0 || nr%divisor!=0) return false;
            ctr = ctr*10;
        } while (ctr<nr);
        return true;
    }
    vector<int> selfDividingNumbers(int left, int right) {
      int ctr = left;
        vector <int> v;
        while (ctr<=right)
        {   
            if (isSelfDividing(ctr)) 
                v.push_back(ctr);
            ctr++;
        }
        return v;
    }
};
__________________________________________________________________________________________________

File: Python3/728.py
__________________________________________________________________________________________________
sample 36 ms submission
class Solution:
    def selfDividingNumbers(self, left, right):
        nums = []
        for n in range(left, right+1):
            if 1<=n<10:
                nums.append(n)
            else:
                is_divide = True
                nn = n
                while n > 0:
                    if n%10 == 0 or nn % (n%10) != 0:
                        is_divide = False
                        break
                    n //= 10
                if is_divide:
                    nums.append(nn)
        return nums
__________________________________________________________________________________________________
sample 12944 kb submission
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        ans = []
        for i in range(left, right + 1):
            shift = i
            is_self_dividing = True
            while(shift != 0):
                ones_digit = shift % 10
                if ones_digit == 0 or i % ones_digit != 0:
                    is_self_dividing = False
                    break
                shift = (shift - ones_digit) // 10
            if is_self_dividing:
                ans.append(i)
        return ans
__________________________________________________________________________________________________
